### PR TITLE
[Issue-152] Drop travis build and use github actions for CI pipeline

### LINKED ISSUE #152: Drop travis build and use github actions for CI pipeline
**Problem description**
travis-ci.org will be shutting down in several weeks, with all accounts migrating to travis-ci.com.
As Pravega has already transitioned to use github actions in pravega/pravega#5563.
We need to do this as well.

**Problem location**
`.travis.yml`

**Suggestions for an improvement**

### PR DESCRIPTION
**Pull request by thekingofcity**
Signed-off-by: zhongle.wang <zhongle_wang@dell.com>

**Change log description**

* Delete .travis.yml as we will no longer use Travis CI.
* Set up GitHub Actions workflow as follows:
    * Build and run tests on every push and pull requests.
    * Publish the snapshot **ONLY** on the push event of the master branch.
* Bonus features:
    * Gradle packages are cached, *but for for now we see no speed up ):*.
    * Workflow could be manually triggered via the Github Actions page, as long as the workflow file is in the master or default branch.

**Purpose of the change**

Fixes #152 

**What the code does**

When there is a push or a pull request, trigger the ci/cd to build and run tests.
The process will be:
* Clone the repository.
* Setup the JDK.
* Get the cache, or download the new packages if there is a update in `gradle.properties`.
* Grant execute permission for gradlew.
* Build via Gradle
* Publish the snapshot if it is a push on the master.

**How to verify it**

Please verify that this is functionally equivalent to the previous CI pipeline:

* It compiles the code.
* It executes all unit tests and all integration tests.
* It uploads code coverage correctly.
* For master and release branches, it publishes to JFrog artifactory **ONLY** on push event.
* Triggers are correct and make sense.

### CODE REVIEW

**Comment by thekingofcity** [comment]
We also have to make sure secrets `BINTRAY_KEY` and `BINTRAY_USER` are added in Github Actions secrets [pravega/hadoop-connectors](https://github.com/pravega/hadoop-connectors/settings/secrets/actions). These secrets are used to publish the snapshot.

**Comment by thekingofcity** [comment]
The [latest workflow result](https://github.com/thekingofcity/hadoop-connectors/actions/runs/660507184) of this file.

File: .github/workflows/build-artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        run: chmod +x gradlew
+
+      - name: Build via Gradle
+        run: ./gradlew clean build
+
+  snapshot:
+    needs: [build]
+    runs-on: ubuntu-latest
+    # only publish the snapshot when it is a push on the master or the release branch (starts with r0.x or r1.x)
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
```
Comment (crazyzhou) [inline .github/workflows/build-artifacts.yml]: We also need to publish `r0.x` and `r1.x` artifacts here, you can add back the checks

**Review by crazyzhou — CHANGES_REQUESTED** [review]

**Comment by crazyzhou** [comment]
@fpj Please help to review and reset the validation checks, thanks

File: .github/workflows/build-artifacts.yml (hunk 1 above)
Comment (thekingofcity) [reply .github/workflows/build-artifacts.yml]: Fixed. I removed this constraint because it is not required in the `.travis.yml`.

**Review by crazyzhou — APPROVED** [review]
LGTM

**Pull request merged**